src/main/java/concurrency/ClassExtendThread.java

```diff
@@ -0,0 +1,17 @@
+package concurrency;
+
+import org.apache.log4j.Logger;
+
+public class ClassExtendThread extends Thread {
+    private static final Logger LOGGER = Logger.getLogger(ClassExtendThread.class);
+
+    @Override
+    public void run() {
+        int value = 0;
+        for (int i = 0; value <= 100; i++) {
+            value = Counter.getCount() + 1;
+            Counter.setCount(value);
```

Comment:
There is too much stuff going on here. Make value processing through Counter's `increment()` method, it will simplify the flow